_posts/2022-09-08-storage-write-removal-before-conditional-termination.md

```diff
@@ -37,6 +37,8 @@ More concretely, the bug can occur if a contract contains the following pattern:
     - It overwrites the storage write in (1).
     - It reverts.
 
+If the initial storage write could possibly be read back (directly or in any internal or external call) between (1) and (3), it is, of course, never removed. (Note, however, that in presence of the optimizer it is not always easy to determine if a storage read at the Solidity level will actually translate to a load instruction in assembly. For example the [Load Resolver](https://docs.soliditylang.org/en/latest/internals/optimizer.html#load-resolver) step can use the knowledge of values that were written before by the contract to replace a `sload()` directly with the value that it would read.)
```

Comment:
```suggestion
If the initial storage write could possibly be read back (directly or in any internal or external call) between (1) and (3), it is, of course, never removed. (Note, however, that in presence of the optimizer it is not always easy to determine if a storage read at the Solidity level will actually translate to a load instruction in assembly. For example the [Load Resolver](https://docs.soliditylang.org/en/latest/internals/optimizer.html#load-resolver) step can use the knowledge of values that were written before by the contract to replace a ``sload()`` directly with the value that it would read.)
```

Oh, looks like I used single backticks :)

Technically those are fine too but for some reason we're using double ones on the blog...